### ISSUE TITLE
Feature request: A way to check for empty values

### ISSUE THREAD

**Issue by xuo** [evanshortiss/env-var#83]
As far as i can tell, there currently doesn't seem to be a way to check if a value is empty or not - which seems like a basic feature for a library like this.

Especially when working with `.env.example` files you sometimes run into issues where someone might forget to fill out some values - having a way to ensure that the original value (before type coercion) is not empty, would be awesome.

I've tried using a extraAccessors (e.g. `asStringNotEmpty()`) but the extraAccessors doesn't get called when the property is undefined and extraAccessors are not chainable with `required()`.

I'd be happy to work on a PR for this, but hear some thoughts first as to how to implement this or if you would want to consider implementing this feature at all.

**Comment by evanshortiss** [comment]
@xuo does using `required` not work for your use case?

```js
// This will throw an error if the SOME_URL_STRING variable is not set
const someUrl = env.get('SOME_URL_STRING').required().asUrlString()
```

Maybe you can provide a code example of the behaviour you'd like?

**Comment by xuo** [comment]
Let's say i have a environment file that i copied, but i forgot to fill out a value - so the the key exists but the value is empty

```js
AWS_ACCESS_KEY_ID=
```

This would pass the `.required()` check and return a empty string

```js
// no error
const accessKeyId = env.get('AWS_ACCESS_KEY_ID').required().asString()
```
Ideally `.required()` itself would either check if the string is empty or there would be another optional chainable function after `.required()` like `.notEmpty()` which would throw a appropriate error message

```js
// EnvVarError: env-var: "AWS_ACCESS_KEY_ID" can't be empty
const accessKeyId = env.get('AWS_ACCESS_KEY_ID').required().notEmpty().asString()
```

**Comment by evanshortiss** [comment]
I see what you mean @xuo, thanks for the explanation. 

> Ideally .required() itself would either check if the string is empty

I feel like this is the correct solution. If a variable is required it should be a non-empty (i.e length >= 1) string in `process.env`. Maybe some recent contributors have an opinion:

@todofixthis @shawnmclean @rmblstrp, any objections to `required()` being changed to throw if a variable is defined as an empty string?

**Comment by todofixthis** [comment]
> @todofixthis @shawnmclean @rmblstrp, any objections to `required()` being changed to throw if a variable is defined as an empty string?

That works for me.

I ran into a similar situation working on a Python library that has some functionality similar to env-var.  I created a filter that would reject incoming "empty" values... but all filters allow `null` by default, which caused problems for some users.  I ended up [adding a parameter](https://github.com/todofixthis/filters/blob/2.0.2/filters/simple.py#L526-L530) controlling whether the filter would also reject `null` values, and for convenience, I also created an additional filter filter coincidentally also named `Required` [with this parameter hard-wired to also reject `null`](https://github.com/todofixthis/filters/blob/2.0.2/filters/simple.py#L556-L562).
